src/main/webapp/app/course/learning-paths/components/competency-node/competency-node.component.html

```diff
@@ -0,0 +1,10 @@
+<div class="d-inline-block competency-node">
+    <div class="progress-container d-inline-block" [ngClass]="{ 'green-competency': isMastered(), 'yellow-competency': isStarted(), 'gray-competency': isNotStarted() }">
```

Comment:
These are method calls, aren't they? Could we get rid of them?

Reply:
With the use of signals only it doesn't matter anymore if you call methods within the template. More infos on that here: https://github.com/angular/angular/discussions/49682
![image](https://github.com/ls1intum/Artemis/assets/29427964/3e3df698-a94c-4c87-aaa4-3d4a687db9a9)

Reply:
The RFC you've linked also mentions:

> You can mark a component as a signal-based component by setting signals: true in the component metadata:
>
> ```
> @Component({
>   signals: true,
>   ...
> })

Can you just confirm that the component counts as signal-based even though we do not have this declaration?

Reply:
this has been removed again. Apparently Angular now self detects if only signals are used in the template.